test/e2e/utils/pods/rtepod/rtepod.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// rtepod includes utilities to introspect the RTE pod - and the RTE pod only
+package rtepod
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	RTEExecutable = "resource-topology-exporter"
+
+	notificationFileOption = "--notify-file"
+)
+
+func FindMetricsPort(rtePod *corev1.Pod) (int, error) {
+	for _, envVar := range rtePod.Spec.Containers[0].Env {
```

Comment:
Can we always assume that the first container will be the RTE one? Maybe we can iterate over all containers and use one that have env var `METRICS_PORT`

Reply:
indeed, I overlooked that part. Will fix.